ci/notebooks-digest-updater.sh

```diff
@@ -0,0 +1,117 @@
+#!/bin/bash
+
+USER_HASH=$1
+REPO_OWNER=$2
+BRANCH=$3
+REPO_NAME=$4
+
+# Fetch the latest commit hash (or use the user-provided one)
+fetch_latest_hash() {
+    local org="$1"
+    local api_url="https://api.github.com/repos/$org/$REPO_NAME"
+
+    if [[ -n "$USER_HASH" ]]; then
+        HASH=$USER_HASH
+        echo "Using user-provided HASH: $HASH"
+    else
+        PAYLOAD=$(curl --silent -H 'Accept: application/vnd.github.v4.raw' "$api_url/commits?sha=$BRANCH&per_page=1")
+        HASH=$(echo "$PAYLOAD" | jq -r '.[0].sha' | cut -c1-7)
+        echo "Extracted HASH: $HASH"
+    fi
+}
+
+# Updates the commits in commit.env
+update_commits() {
+    local REPO_ROOT="$1"
+    local HASH="$2"
+    local COMMIT_ENV_PATH="$REPO_ROOT/manifests/base/commit.env"
+
+    # Get the complete list of commits N-version to update
+    local COMMITS
+    COMMITS=$(grep "\-n=" "$COMMIT_ENV_PATH" | cut -d "=" -f 1)
+
+    for val in $COMMITS; do
+        echo "Updating commit '${val}' to $HASH"
+        sed -i "s|${val}=.*|${val}=${HASH}|" "$COMMIT_ENV_PATH"
+    done
+}
+
+REPO_ROOT=$(git rev-parse --show-toplevel)
+MANIFEST_DIR="$REPO_ROOT/manifests/base"
```

Comment:
is this used somewhere in this script? :thinking:

Reply:
nope! Removed it